pyrokinetics/gk_code/GKOutputReaderCGYRO.py

```diff
@@ -260,59 +290,69 @@ def _set_fields(
+                        continue
+                    eig_shape = [2, data.ntheta, data.ntime]
+                    eig_data = raw_eig_data[: np.prod(eig_shape)].reshape(
+                        eig_shape, order="F"
                     )
-                    fields[ifield, :, :, :, :] = 0
+                    eig_data = eig_data[0] + 1j * eig_data[1]
+                    # Get field amplitude
+                    middle_kx = (data.nradial // 2) + 1
+                    field_amplitude = np.real(field_data[middle_kx, 0, 0, :])
+                    # Multiply together
+                    # FIXME We only set kx=ky=0 here, any other values are left undefined
+                    #       as fields is created using np.empty. Should we instead set
+                    #       all kx and ky to these values? Should we expect that nx=ny=1?
+                    fields[:, 0, 0, :] = eig_data * field_amplitude
                     continue
-                eig_shape = [2, data.ntheta, data.ntime]
-                eig_data = raw_eig_data[: np.prod(eig_shape)].reshape(
-                    eig_shape, order="F"
+
+                # Poisson Sum (no negative in exponent to match frequency convention)
+                q = gk_input.get_local_geometry_miller().q
+                for i_radial in range(data.nradial):
+                    nx = -data.nradial // 2 + (i_radial - 1)
+                    field_data[i_radial, :, :, :] *= np.exp(2j * pi * nx * q)
+
+                fields = field_data.reshape(
+                    [data.ntheta, data.nkx, data.nky, data.ntime]
                 )
-                eig_data = eig_data[0] + 1j * eig_data[1]
-                # Get field amplitude
-                middle_kx = (data.nradial // 2) + 1
-                field_amplitude = np.real(field_data[middle_kx, 0, 0, :])
-                # Multiply together
-                # FIXME We only set kx=ky=0 here, any other values are left undefined
-                #       as fields is created using np.empty. Should we instead set
-                #       all kx and ky to these values? Should we expect that nx=ny=1?
-                fields[ifield, :, 0, 0, :] = eig_data * field_amplitude
-                continue
 
-            # Poisson Sum (no negative in exponent to match frequency convention)
-            q = gk_input.get_local_geometry_miller().q
-            for i_radial in range(data.nradial):
-                nx = -data.nradial // 2 + (i_radial - 1)
-                field_data[i_radial, :, :, :] *= np.exp(2j * pi * nx * q)
+            fields *= cgyro_field_units[field_name]
+            fields = fields.to(local_norm.pyrokinetics).magnitude
 
-            fields[ifield, :, :, :, :] = field_data.reshape(
-                [data.ntheta, data.nkx, data.nky, data.ntime]
-            )
+            data[field_name] = (coords, fields)
 
-        data["fields"] = (coords, fields)
+        data = data.pint.quantify(pyro_field_units)
         return data
```

Comment:
Could just return straight away?

```suggestion
        return data.pint.quantify(pyro_field_units)
```